fix(*): vendoring packages

arschles opened the pull request:
relies on https://github.com/deis/pkg/pull/2

arschles committed:
fix(*): vendoring packages

arschles committed:
fix(boot.go,glide.yaml): fix all vendored deps

technosophos commented on glide.yaml:
```diff
@@ -0,0 +1,12 @@
+package: main
+import:
+- package: gopkg.in
```
I think the way this should work is that you should have a separate line for each _project_, like this:

``` yaml
package: gopkg.in/tomb.v1
package: gopkg.in/yaml.v2
```

The easy way to test this would be to just do a `glide get gopkg.in/yaml.v2` and let it build the appropriate record.

The reason for this is that Glide sees _projects_ and _packages_. A project corresponds to the actual repo, and a package is the Go package or packages. Glide calls packages `subpackages` to sort of conceptually delineate that they are inside of a repository.

arschles committed:
fix(glide.yaml): listing gopkg.in dependencies correctly

arschles commented on glide.yaml:
```diff
@@ -0,0 +1,10 @@
+- package: gopkg.in/olivere/elastic.v2
+- package: gopkg.in/tomb.v1
+- package: gopkg.in/v2/yaml
+- package: gopkg.in/yaml.v2
```
@technosophos glide gets these dependencies correctly now. thanks

arschles committed:
fix(glide.yaml): removing unneeded, massive deis/deis dep

mboersma commented:
This could also use a `make bootstrap` target to get `glide` up to date:

``` Makefile
bootstrap:
    glide -y glide.yaml up
```

But when I do that I get errors about the `protobuf` package at Google Code. Maybe this should refer to its new home at https://github.com/google/protobuf instead?

``` console
...
[INFO] Fetching updates for code.google.com/p/protobuf.
[WARN] Unable to checkout code.google.com/p/protobuf
[WARN] Update failed for code.google.com/p/protobuf: Cannot detect VCS
...
```

bacongobbler commented on Makefile:
```diff
@@ -7,23 +7,19 @@ export GO15VENDOREXPERIMENT=1
 build:
 	mkdir -p ${BINDIR}
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -a -installsuffix cgo -ldflags '-s' -o $(BINDIR)/boot boot.go || exit 1
+	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 GO15VENDOREXPERIMENT=1 go build -a -installsuffix cgo -ldflags '-s' -o $(BINDIR)/boot boot.go || exit 1
```
this is already exported on line 3, so it looks like it's redundant?

bacongobbler commented on glide.yaml:
```diff
@@ -0,0 +1,9 @@
+package: main
+import:
+- package: gopkg.in/natefinch/lumberjack.v2
```
we should probably lock these to a known good version. I think glide 0.8.0 will resolve that issue in Masterminds/glide#143 though so use your own discretion here.

arschles committed:
fix(Makefile): adding a bootstrap target and removing redundant GO15VENDOREXPERIMENT

arschles committed:
fix(glide.yaml): adding version numbers

arschles commented:
@mboersma I added a `bootstrap` build target in the Makefile, and also changed the protobuf dependency. Also, @bacongobbler I locked down version numbers in https://github.com/deis/minio/commit/902e8f71dbd1dc2087f151a675b0cd7b1488f85b

bacongobbler commented:
looks like `rootfs/bin/boot` was accidentally committed in as well. That should get removed :)

arschles commented:
@bacongobbler removed

mboersma commented on Makefile:
```diff
@@ -7,20 +7,19 @@ export GO15VENDOREXPERIMENT=1
 all: build docker-build docker-push
 
+bootstrap:
+	glide -y glide.yaml up
```
Since `glide.yaml` is the default, this could be simplified to just `glide up`.

technosophos commented on Makefile:
```diff
@@ -7,20 +7,19 @@ export GO15VENDOREXPERIMENT=1
 LDFLAGS := "-s -X main.version=${VERSION}"
 BINDIR := ./rootfs/bin
-DEV_REGISTRY ?= $(eval docker-machine ip deis):5000
+DEV_REGISTRY ?= $(docker-machine ip deis):5000
```
Shouldn't this be `$(shell docker-machine...)`? IIRC, there are cases where dm may need env info.

https://www.gnu.org/software/make/manual/html_node/Shell-Function.html

I thought I commented on this before, but I can't find that comment.

sgoings replied:
As long as docker-machine exists in your PATH this should work (and we don't need any shell expansion). I worked with @arschles on this fix as it wasn't working as written with the eval before.

krancour replied:
What about on Linux systems that may have docker running natively and don't have docker-machine installed?

See the following for discussion and possible fix:

https://github.com/deis/prototype-repo/issues/8
https://github.com/deis/prototype-repo/pull/11

arschles replied:
@krancour can we continue the discussion at https://github.com/deis/prototype-repo/issues/8 without blocking this PR?

arschles replied:
@technosophos also discussed [here](https://github.com/deis/minio/pull/5#discussion_r45805082)

technosophos replied:
I see. Okay. I am good with this.

technosophos commented:
Given that @mboersma LGTMed, and his is the only other outstanding comment, I'm adding LGTM as well.

arschles committed:
fix(Makefile): removing -y to use default glide file

arschles commented:
thanks @technosophos

FYI @mboersma - I removed the `-y` to just use the default `glide.yaml` file. also, modified the protobuf import to use the github.com version
